### DISCUSSION TITLE
Open .ifc file and write to new file results in incorrect placement of objects

### DISCUSSION THREAD

**Discussion by BrianPainter** [IfcOpenShell/IfcOpenShell#5637] [Q&A]
using ifcopenshell 0.8.0, python 3.11, windows.

View the file in IFC viewer, and the model is displayed correctly
run the following in python script:

```
import ifcopenshell
model = ifcopenshell.open('original.ifc')
model.write('modified.ifc')
```

View the newly written file in IFC viewer, and model geometry is all located incorrectly. Elements have now rotated and moved.
original file is about 19mb

File is here: [original.ifc](https://tindallcorp-my.sharepoint.com/:u:/p/brianpainter/EQ9Bm_OJynZEs9K7y7tWz_cBALhkyD9Qxr3WtqDHyiffIQ?e=HyNERA)

Update: This works correctly in the 0.7.0 version of the package.

**Comment by aothms** [comment]
So, there is indeed a difference in serialization when it comes to 0.7 and 0.8, but I can't reproduce a visual difference. Which software do you use?

There are a couple of serialization flaws in your model:

1. You use integer literals for floating point numbers, this is not allowed:

`#7835=IFCCARTESIANPOINT((0.42703750729560852,0,0));`

IfcOpenShell v0.8 will translate this to:

`#7833=IFCCARTESIANPOINT((-0.427037507295609,0.,0.));`

note the `.` after the `0`. In 10303 p21 an express `real` literal need to have that dot otherwise it's an integer.

2.  Your units contain `$` (omitted) values where `*` (derived in subtype) is required.

`#8=IFCSIUNIT($,.LENGTHUNIT.,$,.METRE.);`

IfcOpenShell v0.8 will translate this to:

`#8=IFCSIUNIT(*,.LENGTHUNIT.,$,.METRE.);`


IfcOpenShell v0.7 had a more lazy parsing mechanism so as long as you didn't touch these values in the in-memory model they would be serialize verbatim as parsed. In v0.8 this lazy mechanism has been removed.

But again I don't spot any difference that would account for the visual discrepancy you describe.
